src/command/subgraph/publish.rs

```diff
@@ -74,4 +126,84 @@ impl Publish {
             publish_response,
         })
     }
+
+    // output stream is optionally injected (defaulting to stdout) so we can test this function
+    pub fn handle_invalid_routing_url(
+        maybe_prompt_response: Option<bool>,
+        maybe_output: Option<&mut dyn std::io::Write>,
+    ) -> Option<RoverOutput> {
+        let default_output = &mut std::io::stdout();
+        let output = maybe_output.unwrap_or(default_output);
+        if let Some(prompt_response) = maybe_prompt_response {
+            if prompt_response {
+                None
+            } else {
+                writeln!(output, "Publish cancelled by user")
+                    .expect("Could not write to provided output stream");
```

Comment:
this expect should be replaced with a `?` and `.context()` or a `.with_context` from the `anyhow` crate when we update this function to return a `RoverResult`